src/components/formio/multiple.tsx

```diff
@@ -0,0 +1,137 @@
+import Description from './description';
+
+/*
+  name, label and description are typically passed in via the generic form.io
+  component properties. We need to enforce them here, as the wrapper layout needs
+  them.
+*/
+export interface CommonInputProps extends ComponentLabelProps {
+  name: string;
+  description?: string;
+}
+
+export interface MultipleProps<P extends {}, T extends unknown> {
+  as: React.ComponentType<P>;
+  defaultValue: T; // data type, which is component type specific (textfield -> string, number -> number)
+}
+
+const Multiple = <P extends {}, T>(props: MultipleProps<P, T> & CommonInputProps) => {
+  const {label, description, required, tooltip} = props;
+  const {as: WrappedComponent, defaultValue, name, ...otherProps} = props;
+  // TypeScript can't infer this itself with generics, as it doesn't know yet what P looks
+  // like by omitting multiple - but *we* know that HoC forward the remainder of the props.
+  const nestedProps = otherProps as unknown as P;
+
+  const {getFieldProps} = useFormikContext();
+  let {value: fieldValue} = getFieldProps<T[]>(name);
+
+  // due to possibly bad value initialization, it's possible the value is undefined or
+  // even a completely different type (because you're switching from multiple false -> true)
+  // TODO: figure out we can figure this at the formik level? this probably depends on render
+  // order though
+  if (!Array.isArray(fieldValue)) {
+    fieldValue = [];
+  }
+
+  return (
+    <RenderContext.Provider value={{bareInput: true}}>
+      {label && <ComponentLabel label={label} required={required} tooltip={tooltip} />}
+      <FieldArray name={name}>
+        {arrayHelpers => (
+          <table className="table table-bordered">
+            <tbody>
+              {(fieldValue || []).map((_, index) => (
+                <tr key={index}>
+                  <td>
+                    <WrappedComponent name={`${name}[${index}]`} {...nestedProps} />
+                  </td>
+                  <td>
+                    <button
+                      className="btn btn-secondary"
+                      type="button"
+                      onClick={() => arrayHelpers.remove(index)}
+                    >
+                      <span className="sr-only">
+                        <FormattedMessage
+                          description="'Remove item' screenreader button text"
+                          defaultMessage="Remove item"
+                        />
+                      </span>
+                      <i className="fa fa-times-circle-o" />
+                    </button>
+                  </td>
+                </tr>
+              ))}
+              <tr>
+                <td colSpan={2}>
+                  <button
+                    type="button"
+                    className="btn btn-primary formio-button-add-another"
+                    onClick={() => arrayHelpers.push(defaultValue)}
+                  >
+                    <i className="fa fa-plus"></i>
```

Comment:
Do we use font awesome's auto accessibility here or do we need to add aria-hidden?

Reply:
we don't, as far as I know. This is just replicating the formio vanilla builder, but it can't hurt to add the aria attribute, so I'll do that. Though I think you are particularly masochistic if you try to use the form builder interface through a screen reader :stuck_out_tongue: